### ISSUE TITLE
[typescript-angular2] Bug: Request Content-Type other than json not allowed

### ISSUE THREAD

**Issue by macjohnny** [swagger-api/swagger-codegen#6588]
##### Description

The changes in https://github.com/swagger-api/swagger-codegen/pull/6454 introduced setting the request content-type from the `consumes` property. 
```
if (consumes != null && consumes.length > 0) {
    headers.set('Content-Type', consumes.filter(item => this.isJsonMime(item)).join(";"));
}
```
However, the content type is limited to `json`-like mime types. Therefore the content-type is set to an empty string if we e.g. upload a file and have
```
let consumes: string[] = [
     'multipart/form-data'
];
```
and since `multipart/form-data` is not a json mime-type, this results in setting and empty string value.
This in turn, results in an invalid content-type header being set, as explained e.g. here: https://github.com/angular/angular/issues/11819

```
Content-Length: 315
Content-Type: , multipart/form-data; boundary=----WebKitFormBoundarys7BwSTvRA7Nf3WwQ
Host: petstore.swagger.io
```

This breaks e.g. the PetStore.uploadFile() API in the petStore example:
https://github.com/swagger-api/swagger-codegen/blob/bd5b5860289682d60b621c1e3a10dd081b3f7275/samples/client/petstore/typescript-angular-v4/npm/api/pet.service.ts#L204

##### Swagger-codegen version

2.3.0

##### Swagger declaration file content or url
See the petstore example 

https://github.com/swagger-api/swagger-codegen/blob/bd5b5860289682d60b621c1e3a10dd081b3f7275/samples/client/petstore/typescript-angular-v4/npm/api/pet.service.ts#L633

##### Related issues/PRs

Maybe this is resolved with #6080 

##### Suggest a fix/enhancement

Do not restrict the conten-type to json mime types.
Let angular detect the content-type automatically from the data.

**Comment by macjohnny** [comment]
@stoetti @keradus @sebastianhaas  could you have a look at this issue?
what do you think about letting angular determine the content-type and simply remove the following code:
```
if (consumes != null && consumes.length > 0) {
    headers.set('Content-Type', consumes.filter(item => this.isJsonMime(item)).join(";"));
}
```

**Comment by macjohnny** [comment]
cc @wing328

**Comment by wing328** [comment]
> However, the content type is limited to json-like mime types. Therefore the content-type is set to an empty string if we e.g. upload a file and have

@macjohnny I think that's a bug that needs to be fix. It should not used only JSON-related MIME types.

The correct logic should be checking If JSON-related MIME is found in `consumes` (or `produces`), it should use that. Otherwise, just use the first one in the list if there's more than 1 defined.

**Comment by macjohnny** [comment]
@wing328 i will file a PR as soon as https://github.com/swagger-api/swagger-codegen/pull/6574 is merged

**Comment by macjohnny** [comment]
this issue would be resolved by merging #6295

**Comment by macjohnny** [comment]
this issue was resolved with #6295

**Comment by wing328** [comment]
@macjohnny are you going to file another PR to fix the following as originally reported in this issue?
```
if (consumes != null && consumes.length > 0) {
    headers.set('Content-Type', consumes.filter(item => this.isJsonMime(item)).join(";"));
}
```
I don't think isJsonMime is used at the moment:
```
grep -R isJsonMime modules/swagger-codegen/src/main/resources/typescript-angular/*
modules/swagger-codegen/src/main/resources/typescript-angular/api.service.mustache:    public isJsonMime(mime: string): boolean {
```

**Comment by macjohnny** [comment]
@wing328 the issue was resolved with the changes introduced in #6295. This is why in #6295 I removed the `isJsonMime()` method. See also the comment by @bedag-moo https://github.com/swagger-api/swagger-codegen/pull/6295#issuecomment-333179969

**Comment by macjohnny** [comment]
@wing328 btw are there already a release schedule for 2.3.0?
